examples/scripts/learning_regularization_parameter_maps_for_tv_reconstruction.py

```diff
@@ -0,0 +1,881 @@
+    n_coils : int
+        Number of coils for the simulation.
+    acceleration_factor : float
+        Acceleration factor for undersampling.
+    noise_variance : float
+        Variance of the Gaussian noise to be added.
+    seed : float
+        seed of the random number generator.
+
+    Returns
+    -------
+    tuple
+        A tuple containing k-space data, forward operator adjoint reconstruction,
+        possibly the pseudo-inverse solution (if n_coils>1) and the target image.
+    """
+    # randomly choose trajectories to define the fourier operator
+    ny, nx = target_image.shape[-2:]
+    recon_matrix = mrpro.data.SpatialDimension(z=1, y=ny, x=nx)
+    encoding_matrix = mrpro.data.SpatialDimension(z=1, y=ny, x=nx)
+
+    traj = mrpro.data.traj_calculators.KTrajectoryCartesian().gaussian_variable_density(
+        encoding_matrix=encoding_matrix, acceleration=acceleration_factor, fwhm_ratio=0.2, seed=seed
+    )
+
+    fourier_operator = mrpro.operators.FourierOp(
+        traj=traj,
+        recon_matrix=recon_matrix,
+        encoding_matrix=encoding_matrix,
+    )
+
+    # generate simualated coil sensitivity maps
+    from mrpro.phantoms.coils import birdcage_2d
+
+    if n_coils > 1:
+        csm = birdcage_2d(n_coils, recon_matrix, relative_radius=0.8)
+        csm_operator = mrpro.operators.SensitivityOp(csm)
+        forward_operator = fourier_operator @ csm_operator
+    else:
+        forward_operator = fourier_operator
+
+    (kdata,) = forward_operator(target_image)
+    kdata = add_gaussian_noise(kdata, noise_variance=noise_variance, seed=seed)
+
+    kdata, target_image = normalize_kspace_data_and_image(kdata, target_image)  # type: ignore[assignment]
+
+    (adjoint_recon,) = forward_operator.H(kdata)
+
+    if n_coils > 1:
+        # compute an approximation of the pseudo-inverse solution
+        # Note that this in general only makes sense if the forward operator can be injective,
+        # i.e. we have multiple coils with acceleration factor < n_coils
+        (pseudo_inverse_solution,) = mrpro.algorithms.optimizers.cg(
+            forward_operator.gram, right_hand_side=adjoint_recon, initial_value=adjoint_recon, max_iterations=16
+        )
+        return kdata, forward_operator, adjoint_recon, pseudo_inverse_solution, target_image
+    else:
+        pseudo_inverse_solution = None
+    return kdata, forward_operator, adjoint_recon, pseudo_inverse_solution, target_image
+
+
+# %% [markdown]
+# Download the data from Zenodo, create the datasets and dataloaders and define
+# some auxiliary functions for displaying the images.
+
+# %% tags=["hide-cell"] mystnb={"code_prompt_show": "Show download and dataloading details"}
+import tempfile
+from pathlib import Path
+
+import zenodo_get
+
+dataset = '15407890'
+
+tmp = tempfile.TemporaryDirectory()  # RAII, automatically cleaned up
+data_folder_tv = Path(tmp.name)
+zenodo_get.zenodo_get([dataset, '-r', 5, '-o', data_folder_tv])  # r: retries
+
+n_images = 28
```

Comment:
please add a comment here

Reply:
why is the trainingdataset reduced in size?
wouldnt it be better to use all the images in the dataset?
and use the last 8 images as validation, all others as training?